tests/integration/test_section_page_collision.py

```diff
@@ -0,0 +1,96 @@
+"""Test if section indexes avoid pages."""
+
+import io
+import os
+
+import pytest
+
+import nikola.plugins.command.init
+from nikola import __main__
+from nikola.utils import makedirs
+
+from .helper import append_config, cd
+from .test_empty_build import (  # NOQA
+    test_archive_exists,
+    test_avoid_double_slash_in_rss,
+    test_check_files,
+    test_check_links,
+    test_index_in_sitemap,
+)
+
+
+def test_section_index_avoidance(build, output_dir):
+    """Test section index."""
+
+    def _make_output_path(dir, name):
+        """Make a file path to the output."""
+        return os.path.join(dir, name + ".html")
+
+    sec1 = os.path.join(output_dir, "sec1")
+    colliding = os.path.join(output_dir, "sec1", "post0")
+
+    # Do all files exist?
+    assert os.path.isfile(_make_output_path(sec1, "index"))
+    assert os.path.isfile(_make_output_path(colliding, "index"))
+
+    # Is it really a page?
+    with io.open(os.path.join(sec1, "index.html"), "r", encoding="utf-8") as fh:
+        page = fh.read()
+
+    assert "This is Page 0" in page
+    assert "This is Post 0" not in page
+
+
+@pytest.fixture(scope="module")
+def build(target_dir):
+    """
+    Add subdirectories and create a post in section "sec1" and a page
+    with the same URL as the section index.
+
+    It also enables post sections.
```

Comment:
[Sections were a feature that was dropped in Nikola v8.](https://getnikola.com/blog/upgrading-to-nikola-v8.html#sections-were-replaced-by-categories) This test hasn’t been doing anything interesting for quite a while. That said, the layout created by sections can be recreated by `CATEGORY_DESTPATH_AS_DEFAULT = True; CATEGORY_PAGES_FOLLOW_DESTPATH = True`. It might be useful to activate that with `patch_config` in this test and rename it.

Reply:
Nikola failed me with the various attempts I had a re-enabling the configuration and make the test do something useful.

Do you have any objections in dropping this test module alltogether?

Reply:
```
$ tree posts
posts
├── 1.rst
├── a
│   ├── 1.rst
│   └── 2.rst
├── b
│   └── 1.rst
└── b.rst
```

conf.py changes

```python
CATEGORY_ALLOW_HIERARCHIES = True
CATEGORY_DESTPATH_AS_DEFAULT = True
CATEGORY_DESTPATH_TRIM_PREFIX = True
CATEGORY_PAGES_FOLLOW_DESTPATH = True
```

results:
* `/output/posts/a/index.html` is a category index (“Posts about a”) with two posts
* `/output/posts/b/index.html`… crashes with current master, so that needs fixing

I’ll take it from there, fixing both the code and the test, or dropping it if it’s impossible.